kong/observability/logs.lua

```diff
@@ -150,11 +174,10 @@ function _M.maybe_push(stack_level, attributes, log_level, ...)
 
   -- return if log buffer is full
   if #log_buffer >= max_logs then
```

Comment:
The `log_buffer` could be `request_log_buffer`, and all requests share the same `notified_buffer_full`, so a sticky request might make other good requests lose logs that shouldn't be lost.

Not sure if this is expected or not. Shall we ask PM?

Reply:
by "logs" in this case you probably mean the notification that the buffer is full (or resumed accepting entries), because that is the only thing that is controlled by `notified_buffer_full`, right? In that case I think it's fine, what might happen is that multiple requests exceed the buffer size and the notification is only logged once (which I think is the desired behavior).

Reply:
Ah, yes, you are right, we will still have too many warning logs if keep an independent flag for each individual request.